docs/changes/units/17121.api.rst

```diff
@@ -0,0 +1,2 @@
+The ``UnitBase.in_units()`` method is deprecated.
```

Comment:
I think for the message, one can give a bit more explanation, which makes immediately clear that the reader does not have to worry about this. E.g.,
```
The ``UnitBase.in_units()`` method, which was supported as an
alternative to ``to()`` for backward compatibility with the units code
originally taken from pynbody, is now deprecated.
```

Reply:
> ... which makes immediately clear that the reader does not have to worry about this.

We are not advertising this method in our documentation, so I expect it to have very few users, if any. Those who don't use this method don't need to worry about its deprecation at all. The few who might be using it should find the remark about `to()` being a drop-in replacement to be helpful enough.

Reply:
OK, happy to just keep it as is.